### PR TITLE
improve action creators

### LINKED ISSUE #414: Improve action creators

### LINKED ISSUE #458: Visited event does not open, when you are logged as a user

### PR DESCRIPTION
**Pull request by MaxYashan1899**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/EventsExpress/EventsExpress/projects/2)

## Code reviewers

- [ ] @iuriikhrystiuk 

### Second Level Review

- [x] @sand0 

## Summary of issue

improve action creators

## Summary of change

1. Refactored action-creators
2. Created folder structure for Actions

## Testing approach

1.Create, edit, delete Category
2.Create, edit, delete UnitOfMeasoring
3.Edit profile
4.Create, edit, delete Inventory

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

File: EventsExpress/ClientApp/src/components/SelectNotificationTypes/SelectNotificationType.js [hunk 1]
```diff
@@ -27,6 +29,10 @@ class SelectNotificationType extends Component {
                         component={CheckboxGroup}
                         options={items}                        
                     />
+                    {
+                         error &&
+                        <ErrorMessages error={error} className="text-center" />
```
Comment (sand0) [inline EventsExpress/ClientApp/src/components/SelectNotificationTypes/SelectNotificationType.js]: fix formatting, pls

File: EventsExpress/ClientApp/src/components/contactUs/contactUs.js [hunk 2]
```diff
@@ -1,40 +1,41 @@
...
-                <div className="box text text-2 pl-md-4 " >
-                    <div className="text-left mb-2">Problem Type</div>
-                    <Field name='type' className="form-control" component="select">
-                        <option value="newCategory">New Category</option>;
+                <div className='notfound'>
+                    <form className="notfound-404" onSubmit={this.props.handleSubmit}>
+                        <div className="box text text-2 pl-md-4 " >
+                            <div className="text-left mb-2">Problem Type</div>
+                            <Field name='type' className="form-control" component="select">
+                                <option value="newCategory">New Category</option>;
```
Comment (sand0) [inline EventsExpress/ClientApp/src/components/contactUs/contactUs.js:25]: fix formatting, pls

File: EventsExpress/ClientApp/src/containers/categories/category-add.js [hunk 3]
```diff
@@ -1,65 +1,59 @@
...
+    set_edited_category
+} from '../../actions/category/category-add-action';
 import CategoryEdit from "../../components/category/category-edit";
 
 
 class CategoryAddWrapper extends React.Component {
 
     submit = values => {
-        this.props.add({ ...values });
+      return this.props.add({ ...values });
```
Comment (sand0) [inline EventsExpress/ClientApp/src/containers/categories/category-add.js]: fix formatting, pls

File: EventsExpress/ClientApp/src/containers/categories/category-item.js [hunk 4]
```diff
@@ -16,52 +15,51 @@ class CategoryItemWrapper extends Component {
         if (values.name === this.props.item.name) {
             this.props.edit_cansel();
         } else {
-            this.props.save_category({ ...values, id: this.props.item.id });
+          return this.props.save_category({ ...values, id: this.props.item.id });
```
Comment (sand0) [inline EventsExpress/ClientApp/src/containers/categories/category-item.js]: fix formatting, pls

File: EventsExpress/ClientApp/src/containers/editProfileContainers/editGenderContainer.js [hunk 5]
```diff
@@ -1,25 +1,19 @@
 ﻿import React from "react";
 import EditGender from "../../components/profile/editProfile/editGender";
 import { connect } from "react-redux";
-import edit_Gender from "../../actions/EditProfile/EditGender";
+import edit_Gender from "../../actions/editProfile/gender-edit-action";
 
 class EditGenderContainer extends React.Component {
     submit = value => {
-        this.props.editGender(value);
+       return this.props.editGender(value);
```
Comment (sand0) [inline EventsExpress/ClientApp/src/containers/editProfileContainers/editGenderContainer.js:8]: fix formatting, pls

File: EventsExpress/ClientApp/src/actions/chat/chat-action.js [hunk 6]
```diff
@@ -21,47 +19,49 @@ const api_serv = new ChatService();
...
+                dispatch(setAlert({
+                    variant: 'info',
+                    message: `The event was created which could interested you.`,
+                    autoHideDuration: 5000
+                }));
+            });
+        }
+        catch (err) {
+            console.log('Error while establishing connection :(');
+        }
```
Comment (sand0) [inline EventsExpress/ClientApp/src/actions/chat/chat-action.js:64]: let's implement a retry policy here

File: EventsExpress/ClientApp/src/reducers/event-list.js [hunk 7]
```diff
@@ -1,11 +1,12 @@
-import { event } from '../actions/event-item-view';
+
```
Comment (sand0) [inline EventsExpress/ClientApp/src/reducers/event-list.js]: remove this empty line

**Review by sand0 — APPROVED** [review]

**Review by iuriikhrystiuk — COMMENTED** [review]
All looks good.

**Pull request merged**